### PR TITLE
Bug fixes

### LINKED ISSUE #87: Journal selection mechanism
Recent effort in #79 moved the instrument/source selector to a button/dropdown combination which works well.  We should consider doing the same sort of thing for the journal/data combo box.

Rather than just putting everything into a big list in the dropdown, though, this is a good opportunity to use a better mechanism for displaying the available journals.  For instance, partitioning them into three-year "epochs". To be discussed.

### LINKED ISSUE #98: Text correction in Search menu
The search menu contains an item "Go to specific run number" which is a submenu containing..... "Run Number"!

The item could be renamed to "Locate" perhaps?

### PR DESCRIPTION
**Pull request by EthanDevlinSTFC**
Fixing minor bugs
Improve process efficiency ( filter speed)
Implement button for cycle change
Show series title on hover
closes #87 
closes #98

### CODE REVIEW

File: frontend/chartview.cpp [hunk 1]
```diff
@@ -213,8 +213,8 @@ void ChartView::mouseMoveEvent(QMouseEvent *event)
     }
     else
     {
-        if (hovered_)
-            emit showCoordinates(chart()->mapToValue(event->pos()).x(), chart()->mapToValue(event->pos()).y());
+        if (hovered_ != "")
```
Comment (trisyoungs) [inline frontend/chartview.cpp]:
```suggestion
        if (!hovered_.isEmpty())
```

**Review by trisyoungs — CHANGES_REQUESTED** [review]
Cycle button: :+1:
Filter speed: Actually seems quite slow, particularly given the number of visible columns and data items. Not immediately obvious where the bottleneck is, but is probably worth digging in to a bit.

Unfortunately I can't test the plotting as retrieving selog data seems to be broken:
```
127.0.0.1 - - [14/Feb/2022 14:29:18] "GET /getJournal/sandals/journal_20_3.xml HTTP/1.1" 200 -
127.0.0.1 - - [14/Feb/2022 14:29:20] "GET /getNexusFields/sandals/cycle_20_3/70099 HTTP/1.1" 200 -
[2022-02-14 14:29:24,081] ERROR in app: Exception on /getNexusData/sandals/cycle0_3/70099/:raw_data_1:selog:Prop1 [GET]
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib64/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "backend/isisInternal.py", line 42, in getNexusData
    fieldData = nexusInteraction.fieldData(instrument, cycle, runs, fields)
  File "/home/tris/src/jv2/backend/nexusInteraction.py", line 109, in fieldData
    data.append(runData(nxsFile, fields, run))
  File "/home/tris/src/jv2/backend/nexusInteraction.py", line 58, in runData
    runData.append(runTimes(file))
  File "/home/tris/src/jv2/backend/nexusInteraction.py", line 31, in runTimes
    mainGroup = file['raw_data_1']
TypeError: list indices must be integers or slices, not str
```

**Comment by EthanDevlinSTFC** [comment]
> Cycle button: 👍 Filter speed: Actually seems quite slow, particularly given the number of visible columns and data items. Not immediately obvious where the bottleneck is, but is probably worth digging in to a bit.
> 
> Unfortunately I can't test the plotting as retrieving selog data seems to be broken:
> 
> ```
> 127.0.0.1 - - [14/Feb/2022 14:29:18] "GET /getJournal/sandals/journal_20_3.xml HTTP/1.1" 200 -
> 127.0.0.1 - - [14/Feb/2022 14:29:20] "GET /getNexusFields/sandals/cycle_20_3/70099 HTTP/1.1" 200 -
> [2022-02-14 14:29:24,081] ERROR in app: Exception on /getNexusData/sandals/cycle0_3/70099/:raw_data_1:selog:Prop1 [GET]
> Traceback (most recent call last):
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
>     response = self.full_dispatch_request()
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
>     rv = self.handle_user_exception(e)
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
>     reraise(exc_type, exc_value, tb)
>   File "/usr/lib64/python3.6/site-packages/flask/_compat.py", line 39, in reraise
>     raise value
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
>     rv = self.dispatch_request()
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
>     return self.view_functions[rule.endpoint](**req.view_args)
>   File "backend/isisInternal.py", line 42, in getNexusData
>     fieldData = nexusInteraction.fieldData(instrument, cycle, runs, fields)
>   File "/home/tris/src/jv2/backend/nexusInteraction.py", line 109, in fieldData
>     data.append(runData(nxsFile, fields, run))
>   File "/home/tris/src/jv2/backend/nexusInteraction.py", line 58, in runData
>     runData.append(runTimes(file))
>   File "/home/tris/src/jv2/backend/nexusInteraction.py", line 31, in runTimes
>     mainGroup = file['raw_data_1']
> TypeError: list indices must be integers or slices, not str
> ```

I've noticed the filtering issues and am working towards debugging that
I believe the plotting is working in the spectra pr

**Comment by EthanDevlinSTFC** [comment]
> Cycle button: 👍 Filter speed: Actually seems quite slow, particularly given the number of visible columns and data items. Not immediately obvious where the bottleneck is, but is probably worth digging in to a bit.
> 
> Unfortunately I can't test the plotting as retrieving selog data seems to be broken:
> 
> ```
> 127.0.0.1 - - [14/Feb/2022 14:29:18] "GET /getJournal/sandals/journal_20_3.xml HTTP/1.1" 200 -
> 127.0.0.1 - - [14/Feb/2022 14:29:20] "GET /getNexusFields/sandals/cycle_20_3/70099 HTTP/1.1" 200 -
> [2022-02-14 14:29:24,081] ERROR in app: Exception on /getNexusData/sandals/cycle0_3/70099/:raw_data_1:selog:Prop1 [GET]
> Traceback (most recent call last):
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
>     response = self.full_dispatch_request()
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
>     rv = self.handle_user_exception(e)
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
>     reraise(exc_type, exc_value, tb)
>   File "/usr/lib64/python3.6/site-packages/flask/_compat.py", line 39, in reraise
>     raise value
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
>     rv = self.dispatch_request()
>   File "/usr/lib64/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
>     return self.view_functions[rule.endpoint](**req.view_args)
>   File "backend/isisInternal.py", line 42, in getNexusData
>     fieldData = nexusInteraction.fieldData(instrument, cycle, runs, fields)
>   File "/home/tris/src/jv2/backend/nexusInteraction.py", line 109, in fieldData
>     data.append(runData(nxsFile, fields, run))
>   File "/home/tris/src/jv2/backend/nexusInteraction.py", line 58, in runData
>     runData.append(runTimes(file))
>   File "/home/tris/src/jv2/backend/nexusInteraction.py", line 31, in runTimes
>     mainGroup = file['raw_data_1']
> TypeError: list indices must be integers or slices, not str
> ```

Plotting now fixed

**Review by trisyoungs — APPROVED** [review]
Like the addition of the series title to the current position display. :+1:

**Pull request merged**